src/main/java/org/apache/maven/model/jdom/util/JDomUtils.java

```diff
@@ -97,13 +113,12 @@ public static Element insertNewElement(String name, Element root, int index) {
 
     newElement = new Element(name, root.getNamespace());
     newElement.addContent("\n" + indent);
-    int newElementIndex = index == -1 ? calcNewElementIndex(name, root) : index;
-    root.addContent(newElementIndex, newElement);
+    root.addContent(index, newElement);
 
     if (isBlankLineBeforeElement(name, root)) {
```

Comment:
I don't understand this if then else clause. What is the purpose?

Reply:
Insert blank line before the new element or not. Hint: Remove one clause and compare the results of the (failed) integration tests.

Reply:
odd. dunno why I looked left instead of right. sry.